app/Models/Contact.php

```diff
@@ -19,4 +20,46 @@ public function user()
     {
         return $this->belongsTo(User::class);
     }
+
+    const STATE_TO_INVITE = 1;
+    const STATE_FOLLOWED = 2;
+    const STATE_TO_REACT = 3;
+    const STATE_SKIPPED = 4;
+
+    public function getStateAttribute()
+    {
+        if (!$this->hasAccount()) {
+            return self::STATE_TO_INVITE;
+        }
+        if ($this->isFollowed()) {
+            return self::STATE_FOLLOWED;
+        }
+        if ($this->isReacted()) {
+            return self::STATE_SKIPPED;
+        }
+        return self::STATE_TO_REACT;
+    }
+    public function hasAccount()
+    {
+        return !is_null($this->getAccount());
+    }
+    private function getAccount()
+    {
+        return Account::where('provider', $this->source)
+            ->where('provider_id', $this->source_id)->first();
+    }
+    public function getContactAppUser()
+    {
+        return User::where('id', $this->getAccount()->user_id);
```

Comment:
it needs a fetch statement in order to make this query builder to model.

I suggest adding a `->first()`

or using methods like 
```
User::find()
```
will do the job (since we are using id, I guess we can use find method)